AnnService/src/Core/VectorIndex.cpp

```diff
@@ -312,29 +312,16 @@ VectorIndex::DeleteIndex(ByteArray p_meta) {
 ErrorCode
 VectorIndex::MergeIndex(const char* p_indexFilePath)
 {
-    std::string folderPath(p_indexFilePath);
-    if (!folderPath.empty() && *(folderPath.rbegin()) != FolderSep) folderPath += FolderSep;
-    Helper::IniReader iniReader;
-    if (ErrorCode::Success != iniReader.LoadIniFile(folderPath + "indexloader.ini")) return ErrorCode::FailedOpenFile;
-
-    std::shared_ptr<VectorIndex> addIndex = CreateInstance(iniReader.GetParameter("Index", "IndexAlgoType", IndexAlgoType::Undefined),
-        iniReader.GetParameter("Index", "ValueType", VectorValueType::Undefined));
-    if (addIndex == nullptr) return ErrorCode::Fail;
-    if (ErrorCode::Success != addIndex->LoadConfig(iniReader)) return ErrorCode::Fail;
-    if (ErrorCode::Success != addIndex->LoadIndexData(folderPath)) return ErrorCode::Fail;
+    std::shared_ptr<VectorIndex> addIndex;
+    ErrorCode ret = LoadIndex(p_indexFilePath, addIndex);
+    if (ErrorCode::Success != ret) return ret;
 
-    std::shared_ptr<MetadataSet> pMetadata;
-    if (iniReader.DoesSectionExist("MetaData"))
-    {
-        pMetadata.reset(new MemMetadataSet(folderPath + iniReader.GetParameter("MetaData", "MetaDataFilePath", std::string()),
-            folderPath + iniReader.GetParameter("MetaData", "MetaDataIndexPath", std::string())));
-    }
-    if (pMetadata != nullptr) {
+    if (addIndex->m_pMetadata != nullptr) {
 #pragma omp parallel for schedule(dynamic,128)
```

Comment:
we need a mergethreadcount here as parameter to control processor consuming during merge